pkg/github/client/client.go

```diff
@@ -50,6 +52,11 @@ var runnerPerMinuteRate = map[string]float64{
 
 // New instantiates a new GitHub API client.
 func New(ctx context.Context, settings models.Settings) (*Client, error) {
+	if settings.AccessToken == "" {
```

Comment:
Additional check - if no token is provided, return early with `access token is required` and as downstream (user) error